src/main/java/jenkins/scm/api/SCMEvent.java

```diff
@@ -253,13 +322,47 @@ public String toString() {
     }
 
     /**
-     * Annotation used to flag parts of the event data that comes from the event payload itself and consequently
-     * cannot be trusted as state from the SCM.
+     * Annotation used to flag parts of the event data that comes from the event payload itself.
      */
     @Documented
     @Inherited
     @Target({ElementType.METHOD, ElementType.FIELD, ElementType.LOCAL_VARIABLE})
     @Retention(RetentionPolicy.RUNTIME)
-    public @interface Untrusted {
+    public @interface EventData {
+        /**
+         * The trustability of the event data.
+         *
+         * @return The trustability of the event data.
+         */
+        Trustability value() default Trustability.PAYLOAD;
+    }
+
+    /**
+     * The trustability of the event payload data.
+     *
+     * @see #isPayloadTrusted()
+     * @see #payloadTrustability()
+     * @see EventData
+     */
+    public enum Trustability {
+        /**
+         * Signals that the {@link EventData} annotated method's return value or annotated field content is trusted.
+         */
+        TRUSTED(true),
+        /**
+         * Signals that the {@link EventData} annotated method's return value or annotated field content is untrusted.
+         */
+        UNTRUSTED(false),
+        /**
+         * Signals that the {@link EventData} annotated method's return value or annotated field content's is derived
+         * from the event payload. To determine the trustability of the data for a specific event,
+         * consult {@link #isPayloadTrusted()}.
```

Comment:
So subclasses can override `payloadTrustability` for some payloads to declare that their specific payload was, e.g. signed.

There may be portions of the payload that are trusted and other portions that are not. Provider specific event sub-classes can use the `TRUSTED`, `PAYLOAD` and `UNTRUSTED` annotation values to clarify which parts are trusted, which parts come from the payload (which may or may not be trusted) and which parts are just plain untrusted.

This comes into play if you have an event delivery subsystem where you call back to the SCM to say "did you send this event" and the response is yes (so you know the head and revision but nothing else)